# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/FileSystemManagerParent.cpp
# Commit: 256aef51d87b
# Full Hash: 256aef51d87b1ab1733d8d4fb306a03674b6c22b
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 3
# Description:
#   Bug 1786465 - Move TaskQueue ownership from FileSystemManagerParent to FileSystemDataManager; r=dom-storage-reviewers,jesup
#   
#   After this change, we will create only one task queue per origin.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155554
# ==============================================================================

diff -r cfa7bbd206c3 -r 256aef51d87b dom/fs/parent/FileSystemManagerParent.cpp
--- a/dom/fs/parent/FileSystemManagerParent.cpp	Wed Aug 31 09:34:55 2022 +0000
+++ b/dom/fs/parent/FileSystemManagerParent.cpp	Wed Aug 31 09:34:55 2022 +0000
@@ -24,11 +24,9 @@
 namespace mozilla::dom {
 
 FileSystemManagerParent::FileSystemManagerParent(
-    TaskQueue* aTaskQueue, RefPtr<fs::data::FileSystemDataManager> aDataManager,
+    RefPtr<fs::data::FileSystemDataManager> aDataManager,
     const EntryId& aRootEntry)
-    : mTaskQueue(aTaskQueue),
-      mDataManager(std::move(aDataManager)),
-      mRootEntry(aRootEntry) {}
+    : mDataManager(std::move(aDataManager)), mRootEntry(aRootEntry) {}
 
 IPCResult FileSystemManagerParent::RecvGetRootHandleMsg(
     GetRootHandleMsgResolver&& aResolver) {
@@ -121,9 +119,6 @@
 
 FileSystemManagerParent::~FileSystemManagerParent() {
   LOG(("Destroying FileSystemManagerParent %p", this));
-  if (mTaskQueue) {
-    mTaskQueue->BeginShutdown();
-  }
 
   nsCOMPtr<nsISerialEventTarget> target =
       mDataManager->MutableBackgroundTargetPtr();